paws_utils.js

```diff
@@ -15,8 +15,9 @@ function calcNextCollectionInterval(strategy, curUntilMoment, pollInterval) {
     const DELAY_IN_MINUTES = 15;
     const pollIntervalDelay = process.env.paws_poll_interval_delay && process.env.paws_poll_interval_delay <= 900 ? process.env.paws_poll_interval_delay : 300;
     const nowMoment = moment();
+    // If collection start date gt than 30 days, ingest will not accept data, so adjust the date to last 30 days.
     const nextSinceMoment = curUntilMoment.isAfter(nowMoment) ?
-        nowMoment : curUntilMoment;
+        nowMoment : nowMoment.diff(curUntilMoment, 'days') > 30 ? moment().subtract(30, 'days') : curUntilMoment;
```

Comment:
we should be able to use `nowMoment().subtract(30, 'days')`

Reply:
If we used nowMoment() we get the nowMoment is not a function . If i try directly nowMoment.subtract(30, 'days') then nowMoment.diff(nextSinceMoment, 'days') return 0 days instead of 30 days.
So used moment() directly.

Reply:
right, that is because you need a copy of `nowMoment`, like `moment(nowMoment).subtract(30, 'days')`.
Otherwise `moment()` from line 17 and `moment()` from line 20 will be two different timestamps.